jgrapht-core/src/main/java/org/jgrapht/intervalgraph/interval/Interval.java

```diff
@@ -0,0 +1,65 @@
+package org.jgrapht.intervalgraph.interval;
+
+public class Interval<T extends Comparable<T>> implements Comparable<Interval<T>> {
+    private T start;
+    private T end;
+
+    public Interval(T start, T end)  {
+        this.start = start;
+        this.end = end;
+        if (!isValid())
+            throw new RuntimeException();
+    }
+
+    public T getStart() {
+        return start;
+    }
+    public T getEnd() {
+        return end;
+    }
+
+    public boolean isIntersecting(Interval<T> other) {
+        return other.getStart().compareTo(this.getEnd()) <= 0 || this.getStart().compareTo(other.getEnd()) <= 0;
```

Comment:
I think this is wrong. I think this falsely claims that other and this intersect in this example because other ends before this starts:

```
|--other--| |--this--|

```

I think correct would be something like this: `this.contains(other.getStart()) || this.contains(other.getEnd()) || other.contains(this.getStart());`

The last case is just to check if this is completely contained in other, this is why we do not need to check getEnd().

We should add tests that checks all possible relative positions (including matching start and end points as special cases).